central/postgres/pruning.go

```diff
@@ -32,14 +32,28 @@ const (
 		(SELECT 1 FROM clusters WHERE nodes.clusterid = clusters.Id)`
 
 	// Explain Analyze indicated that 2 statements for PLOP is faster than one.
-	deleteOrphanedPLOPDeployments = `DELETE FROM listening_endpoints WHERE processindicatorid in (SELECT id from process_indicators pi WHERE NOT EXISTS
+	deleteOrphanedPLOPDeployments = `DELETE FROM listening_endpoints WHERE poduid IS NULL AND processindicatorid in (SELECT id from process_indicators pi WHERE NOT EXISTS
```

Comment:
so `deleteOrphanedPLOPDeployments` and `deleteOrphanedPLOPPods` are implementing the old process indicator prune logic where if a process indicator was pruned because the deployment or pod was gone then we would also prune the plop.  The existence or not of the poduid does not impact that logic.  I would like to see a scaled up environment with the explain plan of these 2 queries before and after adding the check for `poduid IS NULL` to ensure no negative impact.

Reply:
I ran explain on all four of the queries with and without the null checks. I didn't actually time the queries.

Reply:
I did also measure the timing.

Reply:
so the null check helps.  very good.  Thanks for verifying that.

Reply:
But wait a minute.  The point of this delete is to delete PLOP attached to a process indicator that is going to be pruned.  So those rows should be deleted whether the poduid is populated or not because that pruning is based on the parent process indicator being pruned.  So to maintain the intent of the original query we can't have the null check here.  We would have to ensure that those PLOP are pruned some way before we can justify adding the null check to these 2 delete statemetns.

Reply:
I have removed `poduid IS NULL`. I don't have a strong preference either way. Adding that came from earlier comments where reviewers had a desire to distinguish between old listening endpoints and new listening endpoints.

Reply:
yes I understand that we wanted to make that distinction but that is irrelevant in this query.  I just want to make it clear that this query is deleting based off indicator the plop points to not the poduid of the plop.  So adding `poduid is NULL` changed the intent and result of this statement.

Reply:
Yes. I think we are in agreement now.